PocketSVG.h

```diff
@@ -1,125 +1,40 @@
-//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-//  THE SOFTWARE.
-//
+#import <Foundation/Foundation.h>
+#import <CoreGraphics/CoreGraphics.h>
 
-#import "TargetConditionals.h"
-
-#if TARGET_OS_IPHONE
-#import <UIKit/UIKit.h>
-#else
-#import <Cocoa/Cocoa.h>
-#endif
-
-@interface PocketSVG : NSObject {
-	@private
-	float			pathScale;
-#if TARGET_OS_IPHONE
-	UIBezierPath    *bezier;
-#else
-	NSBezierPath    *bezier;
+#ifdef __cplusplus
+extern "C" {
 #endif
-	CGPoint			lastPoint;
-	CGPoint			lastControlPoint;
-	NSCharacterSet  *separatorSet;
-	NSCharacterSet  *commandSet;
-    
-    NSMutableArray  *tokens;
-}
-
-typedef NS_ENUM(NSInteger, PocketSVGControlPoint) {
-    PocketSVGControlPointInvalid,
-    PocketSVGControlPointS,
-    PocketSVGControlPointQ
-};
-
-@property(nonatomic) PocketSVGControlPoint controlPointType;
-
-#if TARGET_OS_IPHONE
-@property(nonatomic, readonly) UIBezierPath *bezier;
-#else
-@property(nonatomic, readonly) NSBezierPath *bezier;
-#endif
-
 
 /*!
- *  Returns a CGPathRef corresponding to the path represented by a local SVG file's d attribute.
- *
- *  @param nameOfSVG The name of the SVG file. The methods looks for a SVG with the specified in the application's main bundle.
- *
- *  @return A CGPathRef object for the SVG in the specified file, or nil if the object could not be found or could not be parsed.
- */
-+ (CGPathRef)pathFromSVGFileNamed:(NSString *)nameOfSVG;
-
-/*!
- *  Returns a CGPathRef corresponding to the path represented by a local SVG file's D attribute
- *
- *  @param svgFileURL The URL to the file.
- *
- *  @return A CGPathRef object for the SVG in the specified file, or nil if the object could not be found or could not be parsed.
- */
-+ (CGPathRef)pathFromSVGFileAtURL:(NSURL *)svgFileURL;
-
-/*!
- *  Returns a CGPathRef corresponding to the path represented by a string with SVG formatted contents.
+ *  Returns an array of CGPathRefs contained within the passed SVG string.
  *
  *  @param svgString The string containing the SVG formatted path.
+ *  @param attributes An optional pointer for storing a map table containing SVG attributes for the paths
  *
- *  @return A CGPathRef object for the SVG in the string, or nil if no path is found or the string could not be parsed.
- */
-+ (CGPathRef)pathFromSVGString:(NSString *)svgString;
-
-/*!
- *  Returns a CGPathRef corresponding to the path represented by a string with the contents of the d attribute of a path node in an SVG file.
- *
- *  @param dAttribute The string containing the d attribute with the path.
- *
- *  @return A CGPathRef object for the path in the string, or nil if no path is found or the string could not be parsed.
+ *  @return An array of CGPathRef objects or nil if none are found
  */
-+ (CGPathRef)pathFromDAttribute:(NSString *)dAttribute;
-
+NSArray *CGPathsFromSVGString(NSString *svgString, NSMapTable **attributes);
```

Comment:
I've taken a closer look at this, and it turns out this is not really inside a header we need to expose. I'll hide the header and we can leave the method signatures like they are.